runtime/stdlib/random.go

```diff
@@ -91,54 +89,69 @@ func NewRevertibleRandomFunction(generator RandomGenerator) StandardLibraryValue
 		func(invocation interpreter.Invocation) interpreter.Value {
 			inter := invocation.Interpreter
 
-			// TODO: Check if invocation has an argument and implement modulo operation.
-
 			returnIntegerType := invocation.TypeParameterTypes.Oldest().Value
 
+			// arguments should be 0 or 1 at this point
+			var moduloValue interpreter.Value
+			if len(invocation.Arguments) == 1 {
+				moduloValue = invocation.Arguments[0]
+			}
+
+			var randomUint64 uint64
+			if returnIntegerType == sema.UInt8Type || returnIntegerType == sema.UInt16Type ||
+				returnIntegerType == sema.UInt32Type || returnIntegerType == sema.UInt64Type ||
+				returnIntegerType == sema.Word8Type || returnIntegerType == sema.Word16Type ||
+				returnIntegerType == sema.Word32Type || returnIntegerType == sema.Word64Type {
+				randomUint64 = getUint64RandomNumber(generator, returnIntegerType, moduloValue)
+			}
+			var randomBig *big.Int
+			if returnIntegerType == sema.UInt128Type || returnIntegerType == sema.UInt256Type ||
+				returnIntegerType == sema.Word128Type || returnIntegerType == sema.Word256Type {
+				randomBig = getBigRandomNumber(generator, returnIntegerType, moduloValue)
+			}
```

Comment:
Refactor to a `switch` and add a default case which panics with an unreachable error